ts/src/binance.ts

```diff
@@ -7684,7 +7685,27 @@ export default class binance extends Exchange {
         return this.filterByArrayPositions (result, 'symbol', symbols, false);
     }
 
+    async fetchPositionsForSymbol (symbol: string, params = {}) {
+        /**
+         * @method
+         * @name binance#fetchPositions
+         * @description fetch all open positions for specific symbol
+         * @param {string} symbol unified market symbol
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @returns {object[]} a list of [position structure]{@link https://docs.ccxt.com/#/?id=position-structure}
+         */
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        params['type'] = market['type'];
+        params['subType'] = market['linear'] ? 'linear' : 'inverse';
```

Comment:
yes, it's needed, otherwise method would not work, because as it goes through the `fetchPositions` (but with only one specific symbol) it's then comes across this line: https://github.com/ccxt/ccxt/blob/31d2d534b14853edce2aaa94b87949d9914322dd/ts/src/binance.ts#L7825
where final endpoint is decided according to the `type`. (not symbol derived type, but from params).

Reply:
maybe fetchPositions need to be amended to parse if one symbol is provided, however that would be out of the scope of this PR imo.